content/post/2020-01-02-cran-error/index.Rmd

```diff
@@ -0,0 +1,101 @@
+authors:
+  - Julia Romanowska
+date: "`r Sys.Date()`"
+tags:
+- package development
+output: 
+  html_document:
+    keep_md: true
+---
+
+> This post was contributed by [Julia Romanowska](https://jrom.bitbucket.io/homepage/), Researcher at the University of Bergen, Norway. Thank you, Julia!
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(fig.path = "", comment = "")
+
+knitr::knit_hooks$set(
+  plot = function(x, options) {
+    hugoopts <- options$hugoopts
+    paste0(
+      "{{<figure src=",
+      '"', x, '" ',
+      if (!is.null(hugoopts)) {
+        glue::glue_collapse(
+          glue::glue('{names(hugoopts)}="{hugoopts}"'),
+          sep = " "
+        )
+      },
+      ">}}\n"
+    )
+  }
+)
+```
+
+<!--
+  - how we got to know about it
+  - raising an issue on bitbucket(?)
+  - installing docker and checking the docs
+  - searching for R and docker
+  - found r-hub and rocker
+  - running r-hub - problems
+  - figure out where the problem is (docker images didn't have access to the net)
+  - creating a script that runs everything manually, using the image provided by rhub
+-->
+
+I'm involved in development of the [Haplin](https://cran.r-project.org/package=Haplin) R package, which enables fast genetic association analyses (very useful for those involved in genetic epidemiology research). We are a team of scientists that have various background, from genetics, through bioinformatics and statistics. Here's a short text about our latest "adventure" with CRAN and how it taught us some useful stuff.
+
+## OMG, CRAN wrote to me!
+
+There will come the time when you check your mailbox and see this dreaded message about a check that your beloved CRAN package failed. Even though the deadline for submitting a fix is quite reasonable, you might imagine yourself sitting long into the night over your laptop just to find out where is this one comma that was the source of the error. However, I need to say we were positively surprised over the level of details of feedback we got from CRAN concerning the error in _our_ package.
+
+Thus, important :sparkles: tip \#1 :sparkles: *read the e-mail from CRAN carefully!*
+
+## Reproducing the error
+
+OK, let's get to work! Firstly, check the "checks" webpage on CRAN. These CRAN checks come in different _flavors_ and hopefully, not all checks failed. In our case, the flavor that produced the error was `r-devel-linux-x86_64-debian-clang` and I had no wish to install the developmental version of R and other packages on my local laptop just to try mimicking this setup. That's when I thought about checking [*docker*](https://www.docker.com/).
+
+Installing `docker` and running a test went well, but what one needs is a VM that would be exactly the same as the platforms CRAN uses for testing. I realized that it would take too much time to re-create it, so again my DuckDuckGo search engine[^1] came to the rescue.
```

Comment:
```suggestion
Installing `Docker` and running a test went well, but what one needs is a VM that would be exactly the same as the platforms CRAN uses for testing. I realized that it would take too much time to re-create it, so again my DuckDuckGo search engine[^1] came to the rescue.
```